plugin/src/main/java/org/opensearch/ml/sdkclient/RemoteClusterIndicesClient.java

```diff
@@ -226,6 +237,127 @@ public CompletionStage<DeleteDataObjectResponse> deleteDataObjectAsync(
         }), executor);
     }
 
+    @Override
+    public CompletionStage<BulkDataObjectResponse> bulkDataObjectAsync(
```

Comment:
This method seems too long. Can we break down this method?

Reply:
Done.